### ISSUE TITLE
DOC Fix some typos in the docs

### ISSUE THREAD

**Issue by MarcoGorelli** [pandas-dev/pandas#37252]
Here's a few typos I found via `codespell`, first-timers are welcome to open a pull request to fix them
```
doc/source/ecosystem.rst:233: copyed ==> copied
doc/source/whatsnew/v0.24.1.rst:3: Whats ==> What's
doc/source/whatsnew/v0.16.2.rst:150: arithmetics ==> arithmetic
doc/source/whatsnew/v0.24.2.rst:3: Whats ==> What's
doc/source/whatsnew/v1.2.0.rst:183: Expecially ==> Especially
```

**Comment by dsaxton** [comment]
I think that "To wit" isn't a mistake here, I could see changing it because it sounds odd though

**Comment by MarcoGorelli** [comment]
OK have removed it from the issue

**Comment by asharma13524** [comment]
Ok, I believe my latest commit should have addressed this. Please let me know if something is wrong. Thanks.

**Comment by MarcoGorelli** [comment]
closed via #37253
